# HG changeset patch
# User Carlo de Falco <cdf@users.sourceforge.net>
# Date 1354052366 -3600
#      Tue Nov 27 22:39:26 2012 +0100
# Node ID e88b31c485c0a142e1cc1a70bf9e4c1c612744de
# Parent  cd3d7f1261900f6ddd8b763d2f95211f8074c4f8
improve autoconf check for java version and location.

configure.ac: fix issues with checking for java version and location on OSX.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -751,18 +751,18 @@ else
   AC_PATH_PROG(JAVAC, javac, [], [$JAVA_HOME/bin$PATH_SEPARATOR$PATH])
   AC_PATH_PROG(JAR, jar, [], [$JAVA_HOME/bin$PATH_SEPARATOR$PATH])
 fi
 HAVE_JAVA=no
 if test -n "$JAVAC" -a -n "$JAR"; then
   AC_MSG_CHECKING([for Java version])
   java_version=[`$JAVA -version 2>&1 | sed -n -e 's/^java version[^0-9"]*"\([^"]*\)"/\1/p'`]
   AC_MSG_RESULT($java_version)
-  java_major=[`echo -n $java_version | sed -e 's/^\([0-9][0-9]*\)\.\([0-9][0-9]*\)\..*$/\1/'`]
-  java_minor=[`echo -n $java_version | sed -e 's/^\([0-9][0-9]*\)\.\([0-9][0-9]*\)\..*$/\2/'`]
+  java_major=[`echo $java_version | sed -e 's/^\([0-9][0-9]*\)\.\([0-9][0-9]*\)\..*$/\1/'`]
+  java_minor=[`echo $java_version | sed -e 's/^\([0-9][0-9]*\)\.\([0-9][0-9]*\)\..*$/\2/'`]
   if test $java_major -ge 1 -a $java_minor -ge 5; then
     case "$canonical_host_type" in
       *-msdosmsvc*)
         HAVE_JAVA=yes
         JAVA_LIBS=-ladvapi32
       ;;
       *)
         if test -z "$JAVA_HOME"; then
@@ -771,32 +771,33 @@ if test -n "$JAVAC" -a -n "$JAR"; then
           # JavaVMs. Check for the newest version set the JAVA_HOME
           # variable.
           if test -d "/System/Library/Frameworks/JavaVM.framework"; then
             # Sneak the -framework flag into mkoctfile via LFLAGS
             LFLAGS="$LFLAGS -framework JavaVM"
             JAVA_TEMP="/System/Library/Frameworks/JavaVM.framework"
             JAVA_HOME="${JAVA_TEMP}/Home"
             JAVA_ARCH="${JAVA_TEMP}/Libraries/libjvm_compat.dylib"
+            JAVA_LIBS="-framework JavaVM"
             # According to:
             #   http://developer.apple.com/unix/crossplatform.html
             # you need to explicitely set the include path
             JAVA_CPPFLAGS="-I${JAVA_HOME}/include"
             HAVE_JAVA=yes
           # This is the Debian default path
           elif test -d "/usr/lib/jvm/default-java"; then
             JAVA_HOME=/usr/lib/jvm/default-java
           # This is the path of java 6 on debian
           elif test -d "/usr/lib/jvm/java-6-sun"; then
             JAVA_HOME=//usr/lib/jvm/java-6-sun
           else
             JAVA_HOME=/usr/lib/jvm
           fi
         fi
-        JAVA_HOME=[`echo -n $JAVA_HOME | sed -e 's|/$||'`]
+        JAVA_HOME=[`echo $JAVA_HOME | sed -e 's|/$||'`]
         if test -z "$JAVA_ARCH"; then
           if test -d "${JAVA_HOME}/jre/lib/i386"; then
             JAVA_ARCH="i386";
           elif test -d "${JAVA_HOME}/jre/lib/amd64"; then
             JAVA_ARCH="amd64"
           elif test -d "${JAVA_HOME}/jre/lib/mac"; then
             JAVA_ARCH="mac"
           elif test -d "${JAVA_HOME}/jre/lib/maci"; then
